master/pkg/searcher/random.go

```diff
@@ -126,6 +127,9 @@ func (s *randomSearch) trialClosed(ctx context, requestID model.RequestID) ([]Op
 		ops = append(ops, NewClose(create.RequestID))
 		s.CreatedTrials++
 		s.PendingTrials++
+	} else if s.PendingTrials == 0 { // even if the number of created trials == MaxTrials there still might be active trials.
+		shutdown := Shutdown{Failure: len(s.Failures) >= s.TrialsRequested}
```

Comment:
I think we should add this code in `trialExitedEarly`.

Let's say we are running last trial and it ends with `trialExitedEarly`. We have two options:
* if exitedReason is related to HP then we go into `if` and create a new trial
* if exitedReason is not related to HP, then we return `nil, nil`. If I am not missing anything `nil, nil` goes to searcher.go and then to searcher.record function, which expects Create or Shutdown operation, so (I assume) `nil` will be ignored. Since this was our last trial, we will never get trial closed and (probably) never shut down the experiment.